gax-httpjson/src/test/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatterTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.httpjson;
+
+import com.google.common.truth.Truth;
+import com.google.protobuf.Field;
+import com.google.protobuf.Field.Cardinality;
+import com.google.protobuf.Option;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProtoMessageRequestFormatterTest {
+  private Field field;
+  private HttpRequestFormatter<Field> formatter;
+
+  @Before
+  public void setUp() {
+    field =
+        Field.newBuilder()
+            .setNumber(2)
+            .setName("field_name1")
+            .addOptions(Option.newBuilder().setName("opt_name1").build())
+            .addOptions(Option.newBuilder().setName("opt_name2").build())
+            .setCardinality(Cardinality.CARDINALITY_OPTIONAL)
+            .build();
+
+    formatter =
```

Comment:
If `formatter` isn't altered during the tests, can we instantiate this in a `private final` field instead?

Reply:
This is stylistic. Many other tests in gax follow the same pattern (no final, for example: [PagingTest.java#L56](https://github.com/googleapis/gax-java/blob/master/gax/src/test/java/com/google/api/gax/rpc/PagingTest.java#L56)). I kind of like this style because for the following reasons.

Unlike any other non-test class, the tests classes are never used by other classes, but rather are the entry point for each of the tests. This means they are never initialized from some other code, never used in some sophisticated multithreaded environment, nothing explicitly depends on them, as result final or non-final does not really make much difference in terms of maintainability of the class. At the same time, making fields non-final lets initializing them in the `setUp()` method, which is idiomatic (this is what the method is for) plus does not clutter the field definition section of the class.